src/sentry/api/endpoints/auth_index.py

```diff
@@ -100,8 +121,17 @@ def put(self, request: Request):
             return Response(status=status.HTTP_401_UNAUTHORIZED)
 
         validator = AuthVerifyValidator(data=request.data)
-        if not validator.is_valid():
-            return self.respond(validator.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        from_superuser_modal = request.data.get("isSuperuserModal")
+
+        has_valid_sso_session = (
+            False if not request.user.is_superuser else has_completed_sso(request, Superuser.org_id)
+        )
```

Comment:
i'm confused, doesn't this break for non superusers?

Reply:
do we have tests for non superuser sso?

Reply:
this test will test this behaviour `test_no_password_no_u2f`. this will just set `has_valid_sso_session` to false which will `return self.respond(validator.errors, status=status.HTTP_400_BAD_REQUEST)` for any non superuser

Reply:
right, is that intended behavior? what about people using this endpoint to `sudo`?

Reply:
via the flowchart you made, does this use case still work?
![Screen Shot 2022-03-30 at 1 04 51 PM](https://user-images.githubusercontent.com/1976777/160921567-80e89ad5-6244-40fc-af91-61a3ac42edd4.png)

Reply:
it is the intended behavior, people using sudo shouldn't hit this without having `validator.is_valid()` being true, so the 400 wouldn't be returned. If they somehow hit this through sudo without a password or u2f, a 400 should be returned

Reply:
the flowchart still works since the redirect is now on the fe if the user has no password

Reply:
![image](https://user-images.githubusercontent.com/22259802/160937095-2ea72716-866e-41ee-8fbe-4d0642094432.png)